### PR TITLE
closes #35 - link(s) to o'reily field guide

### LINKED ISSUE #35: link(s) to o'reily field guide
I recently saw https://www.oreilly.com/ideas/field-guide-to-the-distributed-development-stack
and thought it might be a good idea to link to them in a lot of the areas in this guide. They also have it as a github repo here: https://github.com/odewahn/dds-field-guide

### PR DESCRIPTION
**Pull request by johnscancella**
Added links to various pages in the O'Reily's A Field Guide to the Distributed Development Stack

### CODE REVIEW

**Comment by johnscancella** [comment]
adding @ccostino since it seems like he is now merging updates?

File: pages/environments.md [hunk 1]
```diff
@@ -24,3 +25,9 @@ Usually you'll need at least four environments, each servering a different purpo
     * Production also has a well defined process for deploying new test builds that have been promoted. See the [Build Promotion] ({{site.baseurl}}/build-promotion/) section.
 
 Note: Most (if not all) environments will need the full set of developer tools. Also, each environment should be sandboxed from each other to prevent resources from accidents.
+
+####Monitoring
+Applications mysteriously failing? Disks have filled to capacity for the 100th time? Everything is working...just VERY SLOWLY? Etc. There is a whole host of problems (bugs) that are difficult or impossible to find without good comprehensive monitoring of the environments. 
+A good starting point would be O'Reily's [The Monitoring Infrastructure Is Critical](http://sites.oreilly.com/odewahn/dds-field-guide/ch07.html) which lists lots of tools that can help with this ongoing problem.
+
+<a name="footnote1">1</a>: See [The Entire Application Runs Locally in Development](http://sites.oreilly.com/odewahn/dds-field-guide/ch05.html) from O'Reily
```
Comment (mbland) [inline pages/environments.md]: We've recently voted against using footnotes in other documents. Can you promote this to appear inline?

File: pages/self-provisioning.md [hunk 2]
```diff
@@ -7,3 +7,5 @@ Developers can create and delete systems on demand, with group allocation and qu
 These self-provisioned systems are identical with documented exceptions for network configuration, authentication, etc. 
 All services should be developed using SSL for proper testing. 
 This requires a self-service system to issue certificates, so applications can be developed and tested using the same configuration used in production.
+
+Cloud based environments, be them private or public based, fulfil most if not all of these requirements. To learn more about the cloud as a platform we recommend O'Reily's field guide located at [The Cloud Is the Default Platform] (http://sites.oreilly.com/odewahn/dds-field-guide/ch02.html)
```
Comment (mbland) [inline pages/self-provisioning.md]: "be them" => "be they"

"fulfil" => "fulfill"

**Comment by mbland** [comment]
Looks good, but for a couple of nits noted in line, and one other: s/O'Reily/O'Reilly/g :smile:

**Comment by mbland** [comment]
@ccostino is leading the development of our [Dev Environment Standardization Guide](https://pages.18f.gov/dev-environment-standardization/) (for lack of a better name), but he's also certainly capable of and welcome to merge commits to this one, too. :smile:

**Comment by johnscancella** [comment]
@mbland good to know. I will make the updates and push them so they reflect in the PR.

File: pages/self-provisioning.md (hunk 2 above)
Comment (johnscancella) [inline pages/self-provisioning.md]: that is what I get for editing this in vim.

**Comment by johnscancella** [comment]
@mbland I changed the footnotes, but not sure if it meets your criteria. Could you give it another once over?

File: index.md [hunk 3]
```diff
@@ -5,4 +5,6 @@ title: Introduction
 Best practices are difficult to practice without a good environment in which to practice them. 
 Here we enumerate the elements of a productive development environment.
 
+O'Reily has also put together a great document detailing a lot of the ideas covered here. You can find it at [A Field Guide to the Distributed Development Stack](http://sites.oreilly.com/odewahn/dds-field-guide/ch01.html)
```
Comment (mbland) [inline index.md]: Missed an O'Reilly.

**Comment by mbland** [comment]
Looks good; ready to merge after one final tweak!

File: index.md (hunk 3 above)
Comment (johnscancella) [inline index.md]: DOH!

**Pull request merged**